app/middleware/authenticate.js

```diff
@@ -8,7 +8,13 @@ function middleware_authenticate (config) {
     // Authentication Middleware
     return function (req, res, next) {
       if (!req.session.loggedIn) {
-        res.redirect(403, '/login');
+        if (config.googleoauth === true) {
+          res.redirect('/login');
+        }
+        else {
+          res.redirect(403, '/login');  
+        }
+        
```

Comment:
Ah, you're right, I've put it in in this case as well (will commit ASAP I address all these issues you've mentioned here). It must have slipped my attention when I was adding the if/else.